src/libs/type-utils/lodash-fp-helpers.ts

```diff
@@ -0,0 +1,138 @@
+import _ from 'lodash/fp'
+import { AnyFn, AnyPromiseFn, GenericFn, WrapFn } from 'src/libs/type-utils/general-types'
+
+/**
+ * (async Promise overload) Provides better type flow and ergonomics for the common case of wrapping a
+ * main function (typically a data call that returns a promise) with
+ * additional pass-thru wrapper functions (handlers) to handle things like
+ * errors or retry.  These handlers are expected to be HoF's
+ * (higher order functions) that return a wrapped function with the desired
+ * handling logic which still honors the same function signature of the fn
+ * it is given as it's single argument
+ * @param handlers
+ * @param mainFn
```

Comment:
Without descriptions, are these JSDoc parameters useful? Isn't this information provided by the function signature?

Reply:
updated